core/dispatcher/src/whisk/core/container/WhiskContainer.scala

```diff
@@ -127,7 +128,8 @@ class WhiskContainer(
                 val http = new HttpUtils(connection, host)
                 val (code, bytes) = http.dopost(endpoint, msg, Map(), timeout)
                 Try { connection.close() }
-                Some(code, new String(bytes, "UTF-8"))
+                val returnCode = if (code == -1) ActivationResponse.ContainerError else code
```

Comment:
See below, this is to normalize the HttpClient interface, which yields magic number `-1` with invoker internal error codes.